exercises/binary-search/example.cpp

```diff
@@ -0,0 +1,40 @@
+#include <stdexcept>
+#include "binary_search.h"
+
+namespace binary_search {
+
+size_t find(std::vector<int> const& data, int const value)
+{
+    if(data.empty() || value < data.front() || data.back() < value)    
+    {
+        throw std::domain_error("Value not in input vector!");
+    }
+    
+    size_t lpos {0};
+    size_t rpos {data.size() - 1};
+    size_t mid  {0};
+
+    while (lpos <= rpos)
+    {
+        mid = (lpos + rpos) >> 1;
+
+        if (data.at(mid) == value) 
+        { 
+            return mid; 
+        }
+        else
+        {
+            if (value < data.at(mid)) 
+            {
+                rpos = mid - 1;
+            } 
+            else if (value > data[mid]) 
+            {
+                lpos = mid + 1;
+            }
+        }
```

Comment:
Another thought on style - I've seen a lot of conditionals get out of control, so I can't help wanting to remove the `else`. Something like:
```c++
if (data.at(mid) == value) 
{ 
    return mid; 
}

if (value < data.at(mid)) 
{
    // snip
```
Again a matter of preference; it's perfectly fine as it is, and I'm sure some would find the current way preferable. 

Definitely not worth holding up a merge for something so small. Thanks for your implementation!

Reply:
Thanks, @KevinWMatthews for your comment. This is a good point, so, I unwrapped the nested if/elses and turned these into a series of simple ifs.